source/_posts/2017-03-30-Haptic-feedback-makes-you-vibrate.md

```diff
@@ -0,0 +1,147 @@
+---
+title:  Haptic Feedback makes you vibrate!
+date: 2017-03-30 10:10:57
+tags: ios,haptic,feedback,vibrate
+authorIds: 
+- maco
+categories:
+- iOS
+---
+
+The iPhones could vibrate ever since they were launched. And so did almost all the other phones, even from before smartphones were a thing. But Apple took this function further than anyone else, by introducing the Taptic Engine in the Apple Watch, and after on in many other products, like the new MacBooks and iPhones. The Taptic Engine provides a haptic feedback, more similar to real-life touches. 
+
+Together with iOS 10, Apple introduced an API for the Taptic Engine, which the developers can use to give haptic feedback to their users. To fully experience the haptic feedback you need an iPhone 7, 7 Plus or newer. 
+
+## How to create haptic feedback
+
+Using the new API is super easy, with the `UIFeedbackGenerator` you can create haptic feedback with only two lines of code.
+
+Even though the API is introduced in iOS 10, it's only the iPhone 7 and iPhone 7 Plus that support this. On all previous devices, nothing happens when you try to do that. 
+
+There are 7 different types of haptic feedback a developer can use. The only difference between those 7 types of feedback is the vibration pattern. Some of the are subtle, like the selection change, and other are more powerful, like the notification error. 
+The `UIFeedbackGenerator` has 3 subclasses, `UISelectionFeedbackGenerator`, `UIImpactFeedbackGenerator` and `UINotificationFeedbackGenerator`, and those are the ones a developer should work with. We are showing how to use them further below.
+
+Due to the inherent difficulties of showing vibrations through a blog post, we are using the official videos of each type of feedback from Apple's Human Interface Guidelines. You can also run the code [from this repo](https://github.com/mariusc/Vibrate) on your iPhone 7 or iPhone 7 Plus to try it out.
+
+### Selection
+
+This one can be used when the user has to select from multiple discrete values, as he scrolls through them. For example, it's used in the `UIPickerView`. It feels like a light tap with each selection change. This is not to be used for example when selecting one of 4 radio button choices.
```

Comment:
> as he scrolls through them

The Californian in me says you shouldn't use "he" like this 😄  

You can use "they scroll", but "as he/she scrolls through them" is probably the more professional way.